components/camel-jetty-common/src/main/java/org/apache/camel/component/jetty/JettyHttpComponent.java

```diff
@@ -1142,15 +1194,20 @@ protected CamelServlet createServletForConnector(
         holder.setInitParameter(CamelServlet.ASYNC_PARAM, Boolean.toString(endpoint.isAsync()));
         context.addServlet(holder, "/*");
 
-        File file = File.createTempFile("camel", "");
-        boolean result = file.delete();
-        if (!result) {
-            LOG.error("failed to delete {}", file);
+        String location = endpoint.getFilesLocation();
+        if (location == null) {
+            File file = File.createTempFile("camel", "");
+            boolean result = file.delete();
+            if (!result) {
+                LOG.error("failed to delete {}", file);
```

Comment:
WARN - Also you can use FileUtil.deleteFile from camel-core

Reply:
Actually, if you check well it is not my code, this is the existing code that I simply moved into a if-block